relay-server/src/services/store.rs

```diff
@@ -633,9 +615,22 @@ impl StoreService {
                 .map(|content_type| content_type.as_str().to_owned()),
             attachment_type: item.attachment_type().cloned().unwrap_or_default(),
             chunks: chunk_index,
+            data,
             size: Some(size),
             rate_limited: Some(item.rate_limited()),
-        })
+        };
+
+        if send_individual_attachments {
+            let message = KafkaMessage::Attachment(AttachmentKafkaMessage {
+                event_id,
+                project_id,
+                attachment,
+            });
+            self.produce(KafkaTopic::Attachments, message)?;
+            Ok(None)
+        } else {
+            Ok(Some(attachment))
+        }
```

Comment:
I don't really understand why you produce the message in this function in the case of `send_individual_attachments` but return it to be produced later in the other case.

Reply:
in that case the attachment metadata is attached directly to the event message, I thought this was looking a bit nicer

Reply:
I think this flow could benefit from some comments